src/OpenTelemetry/Trace/SpanSdk.cs

```diff
@@ -827,7 +828,24 @@ private bool IsAttributeValueTypeSupported(object attributeValue)
                 return true;
             }
 
-            // TODO add array support
+            if (attributeValue is IEnumerable enumerable)
+            {
+                Type entryType = null;
+                foreach (var entry in enumerable)
```

Comment:
should we `try{}catch` it? Iterating over custom type may easily throw.

Reply:
Yeah, make sense to be safe here and return `false` for any exceptions.